code/modules/mob/living/carbon/human/human_defines.dm

```diff
@@ -128,3 +128,10 @@
 
 	/// This is the cooldown on suffering additional effects for when shock gets high
 	COOLDOWN_DECLARE(last_shock_effect)
+
+///This shows when the human last fired a shotgun.
+	var/lastshotguntime
```

Comment:
Shouldnt affect only shotguns

Reply:
> Shouldnt affect only shotguns

Regarding this; marines have been juggling literally any weapon they can get their hands on. TL-127s, Grenade Launchers, and so on. It'd be a good idea to have this PR encompass all guns so we can end the issue altogether.